### ISSUE TITLE
Add seed to ReproducibleDataLoader

### ISSUE THREAD

**Issue by daniellepintz** [pytorch/torchtune#131]
We should use `worker_init_fn` to init the seed for every DL worker

More details in PT docs:
https://pytorch.org/docs/stable/notes/randomness.html#dataloader
